### PR TITLE
Improve debugger code 

### PR DESCRIPTION
**Pull request by lankavitharana**
Fix https://github.com/ballerinalang/ballerina/issues/4347
Fix https://github.com/ballerinalang/ballerina/issues/3274

This PR will add worker debugging functionality.

The worker debugging works same as debugging java threads in IDEA.

What happens is debug will only work for one worker, so other workers will run in the background. 

for example
If there are two workers and we have debug points in both of them, then debug will hit only one worker(can't guarantee which will get hit, because it hits the first point which reaches first). 

When hitting a debug point, the debugger will acquire a debug lock. This lock will only get released if the client does the "RESUME" command, for all other commands, "STEP_IN", "STEP_OVER" etc. this debug lock won't get released, hence the debug flow will continue in the same worker. Other workers will run in the background. Debug points in those workers will get ignored.

If the client hits "RESUME" command then debuger will continue execution of all the workers and the worker who reaches a debug point first will get the lock and hit the debug point. Same as before, other workers will run in the background.

### CODE REVIEW

File: modules/ballerina-core/src/main/java/org/ballerinalang/util/debugger/VMDebugManager.java [hunk 1]
```diff
@@ -325,9 +327,70 @@ public void notifyExit() {
...
+                VariableDTO variableDTO = new VariableDTO(localVarInfo.getVariableName(), "Local");
+                if (BTypes.typeInt.equals(localVarInfo.getVariableType())) {
+                    variableDTO.setBValue(new BInteger(fcp.getLongLocalVars()[localVarInfo.getVariableIndex()]));
+                } else if (BTypes.typeFloat.equals(localVarInfo.getVariableType())) {
+                    variableDTO.setBValue(new BFloat(fcp.getDoubleLocalVars()[localVarInfo.getVariableIndex()]));
+                } else if (BTypes.typeString.equals(localVarInfo.getVariableType())) {
+                    variableDTO.setBValue(new BString(fcp.getStringLocalVars()[localVarInfo.getVariableIndex()]));
+                } else if (BTypes.typeBoolean.equals(localVarInfo.getVariableType())) {
+                    variableDTO.setBValue(new BBoolean(fcp.getIntLocalVars()[localVarInfo
+                            .getVariableIndex()] == 1 ? true : false));
```
Comment (anupama-pathirage) [inline modules/ballerina-core/src/main/java/org/ballerinalang/util/debugger/VMDebugManager.java]: Shall we simplify this to ` fcp.getIntLocalVars()[localVarInfo.getVariableIndex()] == 1`

File: modules/ballerina-test/src/test/java/org/ballerinalang/test/utils/debug/ExpectedResults.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+*/
+package org.ballerinalang.test.utils.debug;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Test Util class to hold expected results.
+ *
+ * @since 0.95
```
Comment (anupama-pathirage) [inline modules/ballerina-test/src/test/java/org/ballerinalang/test/utils/debug/ExpectedResults.java]: Fix the version?

File: modules/ballerina-test/src/test/java/org/ballerinalang/test/utils/debug/WorkerResults.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+* under the License.
+*/
+package org.ballerinalang.test.utils.debug;
+
+import org.ballerinalang.util.debugger.dto.BreakPointDTO;
+
+/**
+ * Test Util class to hold expected worker results.
+ *
+ * @since 0.95
```
Comment (anupama-pathirage) [inline modules/ballerina-test/src/test/java/org/ballerinalang/test/utils/debug/WorkerResults.java]: Incorrect version?

**Comment by hasithaa** [comment]
@lankavitharana This PR has conflicts. Can you fix those?

**Pull request merged**